CRM/Certificate/Token/Certificate.php

```diff
@@ -65,16 +71,63 @@ public function prefetch(TokenValueEvent $e) {
     return $resolvedTokens;
   }
 
+  private function getMembershipDates(TokenValueEvent $e): array {
+    $startDate = NULL;
+    $endDate = NULL;
+
+    $contactIds = $e->getTokenProcessor()->getContextValues('contactId');
+    $contactId = (is_array($contactIds) && !empty($contactIds[0])) ? $contactIds[0] : 0;
+
+    $memberships = Membership::get(FALSE)
+      ->addSelect('start_date', 'end_date')
+      ->addWhere('contact_id', '=', $contactId)
+      ->addWhere('status_id', '=', self::MEMBERSHIP_STATUS_CURRENT)
+      ->execute()
+      ->getArrayCopy();
+
+    foreach ($memberships as $membership) {
```

Comment:
Also, I see, we are looping through all the memberships a contact has, eventually settling with the last one.

can we just put a comment, just so it's clear why we are doing this.